### PR TITLE
Develop one to one with component

### PR DESCRIPTION
**Pull request by vincentpierre**

### CODE REVIEW

File: UnitySDK/Assets/ML-Agents/Scripts/Policy/BehaviorParameters.cs [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+        [HideInInspector]
+        public string behaviorName
+        {
+            get { return m_BehaviorName; }
+        }
+
+        public IPolicy GeneratePolicy(Func<float[]> heuristic)
+        {
+            if (m_Model == null || m_UseHeuristic)
```
Comment (chriselion) [inline UnitySDK/Assets/ML-Agents/Scripts/Policy/BehaviorParameters.cs]: Might need to talk about the order here. If you're starting from scratch and don't have a model, you probably want to train (hopefully the communicator is on). As it's written now, you'll get a HeuristicPolicy which probably will raise an exception.

Comment (chriselion) [reply UnitySDK/Assets/ML-Agents/Scripts/Policy/BehaviorParameters.cs]: Maybe just `if(m_UseHeuristic)` at the start, and also return a HeuristicPolicy if no model and no communicator?

Comment (vincentpierre) [reply UnitySDK/Assets/ML-Agents/Scripts/Policy/BehaviorParameters.cs]: In my view, I wanted the Heuristic to have priority over training. But we should discuss that.

Comment (vincentpierre) [reply UnitySDK/Assets/ML-Agents/Scripts/Policy/BehaviorParameters.cs]: This sounds good to me

**Pull request merged**